src/main/kotlin/org/vividus/testsite/api/HttpRequestController.kt

```diff
@@ -0,0 +1,37 @@
+package org.vividus.testsite.api
+
+import com.github.benmanes.caffeine.cache.Cache
+import com.github.benmanes.caffeine.cache.Caffeine
+import java.util.concurrent.TimeUnit
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.vividus.testsite.api.Cache.cache
+
+@RestController
+class HttpRequestController {
+    @GetMapping("/api/delayed-response")
+    fun delayResponse(@RequestParam clientId: Long): ResponseEntity<Unit> {
+        val firstRequestTimestamp = cache.getIfPresent(clientId)
+        if (firstRequestTimestamp == null) {
+            cache.put(clientId, System.currentTimeMillis())
+            return ResponseEntity.notFound().build()
+        }
+        if (countAgeInSeconds(firstRequestTimestamp) < 5) {
+            return ResponseEntity.notFound().build()
+        }
+        cache.invalidate(clientId)
+        return ResponseEntity.ok().build()
+    }
+
+    private fun countAgeInSeconds(firstRequestTimestamp: Long): Long {
+        return (System.currentTimeMillis() - firstRequestTimestamp) / 1000
+    }
+}
+
+object Cache {
+    val cache: Cache<Long, Long> = Caffeine.newBuilder()
```

Comment:
Just in case why not use https://spring.io/guides/gs/caching/ spring-based caching?

Reply:
I just was told to use a caffeine cache

Reply:
Please ignore the comment, default realization doesn't support flexible eviction policies;